be/src/exprs/vectorized/array_functions.tpp

```diff
@@ -101,4 +102,127 @@ private:
         dest_offsets.emplace_back(dest_offsets.back() + hash_set->size() + has_null);
     }
 };
+
+template <PrimitiveType PT>
+class ArraySort {
+public:
+    using ColumnType = RunTimeColumnType<PT>;
+
+    static ColumnPtr process(FunctionContext* ctx, const Columns& columns) {
+        DCHECK_EQ(columns.size(), 1);
+
+        size_t chunk_size = columns[0]->size();
+        // TODO: For fixed-length types, you can operate directly on the original column without using sort index,
+        //  which will be optimized later
+        std::vector<uint32_t> sort_index;
+        ColumnPtr src_column = ColumnHelper::unpack_and_duplicate_const_column(chunk_size, columns[0]);
+        ColumnPtr dest_column = src_column->clone_empty();
+
+        if (src_column->is_nullable()) {
+            const auto& src_nullable_column = *down_cast<const NullableColumn*>(src_column.get());
+            const auto& src_data_column = src_nullable_column.data_column_ref();
+            const auto& src_null_column = src_nullable_column.null_column_ref();
+
+            auto* dest_nullable_column = down_cast<NullableColumn*>(dest_column.get());
+            auto* dest_data_column = dest_nullable_column->mutable_data_column();
+            auto* dest_null_column = dest_nullable_column->mutable_null_column();
+
+            if (src_column->has_null()) {
+                dest_null_column->get_data() = src_null_column.get_data();
```

Comment:
is std::move here safe?

Reply:
It's not std::move, it's value copy: std::vector::operator=

https://www.cplusplus.com/reference/vector/vector/operator=/